Mathlib/Analysis/SpecialFunctions/MulExpNegSq.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2024 Jakob Stiefel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Stiefel
+-/
+import Mathlib.Data.Complex.ExponentialBounds
+import Mathlib.Topology.ContinuousMap.Bounded.Basic
+
+/-!
+# Definition of `mulExpNegSq`, `mulExpNegMulSq` and properties
+
+The mapping mulExpNegMulSq transforms a function `g` into another function
+`mulExpNegMulSq g ε : fun x => g x * Real.exp (-ε * g x * g x)`.
+
+## Main Properties
+
+- `abs_mulExpNegMulSq_le`: For fixed `ε > 0`, the mapping `mulExpNegMulSq g ε` is bounded by
+  `Real.sqrt ε⁻¹` uniformly in `g`;
+- `tendsto_mulExpNegMulSq`: For fixed function `g`, the mapping `mulExpNegMulSq g ε`
+  converges pointwise to `g` as `ε → 0`;
+- `continuous_mulExpNegMulSq`: If the function `g` is continuous, the function `mulExpNegMulSq g ε`
+  is continuous;
+- `abs_mulExpNegMulSq_le_norm`: For a fixed bounded continuous function `g`, the mapping
+  `mulExpNegMulSq g ε` is bounded by `norm g`, uniformly in `ε ≥ 0`;
+-/
+
+open NNReal ENNReal BoundedContinuousFunction Filter
+
+namespace Real
+
+section mulExpNegSq
+
+/-! ### Definition and properties of `fun x => x * Real.exp (- x * x)`-/
+
+/-- Definition of `mulExpNegSq` -/
+noncomputable
+def mulExpNegSq (x : ℝ) := x * exp (- x * x)
+
+theorem mulExpNegSq_apply (x : ℝ) : mulExpNegSq x = x * exp (- x * x) :=  rfl
+
+theorem neg_mulExpNegSq_neg (x : ℝ) : - mulExpNegSq (-x) = mulExpNegSq x  := by
+  simp only [mulExpNegSq, neg_mul, neg_neg, mul_neg]
+
+theorem mulExpNegSq_le_one (x : ℝ) : mulExpNegSq x ≤ 1 := by
+  rw [mulExpNegSq_apply, neg_mul, ← le_div_iff₀ (exp_pos (-(x * x))), exp_neg,
+    div_inv_eq_mul, one_mul]
+  apply le_trans _ (add_one_le_exp (x * x))
+  nlinarith
+
+theorem abs_mulExpNegSq_le_one (x : ℝ) : |mulExpNegSq x| ≤ 1 := by
+  apply abs_le.mpr ⟨_, mulExpNegSq_le_one x⟩
+  rw [← neg_mulExpNegSq_neg, neg_le_neg_iff]
+  exact mulExpNegSq_le_one (-x)
+
+theorem deriv_expNegSq (y : ℝ) :
+    deriv (fun (x : ℝ) => exp (-x * x)) y = -2 * y * exp (-y * y) := by
+  simp only [neg_mul, differentiableAt_neg_iff, differentiableAt_id', DifferentiableAt.mul,
+    _root_.deriv_exp, deriv.neg', deriv_mul, deriv_id'', one_mul, mul_one, neg_add_rev]
+  ring
+
+theorem expNegSq_differentiableAt (y : ℝ) :
```

Comment:
Our naming convention is that the property of the function should be placed first, so that name should be `differentiableAt_expNegSq`. Could you change this and other lemmas accordingly?